communication/src/coap_channel.h

```diff
@@ -218,25 +249,37 @@ class __attribute__((packed)) CoAPMessage
 	bool prepare_retransmit(system_tick_t now)
 	{
 		CoAPType::Enum coapType = CoAP::type(get_data());
-		if (coapType==CoAPType::CON) {
-			timeout = now + transmit_timeout(transmit_count);
-			transmit_count++;
-			return transmit_count <= MAX_RETRANSMIT+1;
+		if (coapType == CoAPType::CON) {
```

Comment:
While this technically is closer to the spec (taking the timeout with a pre-applied random factor and doubling it vs doubling the timeout and then applying a random factor), the difference in practice is not significant, especially considering the device service uses an entirely different set of parameters.  Both approaches are within the operational constraints of how long the server will wait since the maximum timeout ends up being the same.

I'd like to drive this change from observed problems rather than from a strict adherence to the spec since this is a non-trivial change, so I want to be sure the gain outweighs the risk.  

In terms of gain, I appreciate the inline references to the spec, which adds clarity.  Also welcome other voices here about which is easiest to comprehend since that is a key factor here too. I am on the fence, but can imagine others might find this easier to grasp.